lua/telescope/utils.lua

```diff
@@ -170,6 +170,81 @@ utils.filter_symbols = function(results, opts, post_filter)
   end
 end
 
+utils.path_filename_first = (function()
+  return function(path, path_display)
+    local reverse_directories = false
+
+    if type(path_display["filename_first"]) == "table" then
+      local filename_first_opts = path_display["filename_first"]
+
+      if filename_first_opts.reverse_directories == nil or filename_first_opts.reverse_directories == false then
+        reverse_directories = false
+      else
+        reverse_directories = filename_first_opts.reverse_directories
+      end
+    end
+
+    local dirs = vim.split(path, utils.get_separator())
+    local filename
+
+    if reverse_directories then
+      dirs = utils.reverse_table(dirs)
+      filename = table.remove(dirs, 1)
+    else
+      filename = table.remove(dirs, #dirs)
+    end
+
+    local tail = table.concat(dirs, utils.get_separator())
+    local transformed_path = vim.trim(filename .. " " .. tail)
+    local path_style = { { { #filename, #transformed_path }, "TelescopeResultsComment" } }
+
+    -- Trim prevents a top-level filename to have a trailing white space
+    return transformed_path, path_style
+  end
+end)()
+
+utils.path_truncate = (function()
+  local calc_result_length = function(truncate_len)
```

Comment:
could still be just a top level local function, no need for this function scoping here

Reply:
Moved back to its initial position.